src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -253,6 +254,157 @@ public static StringValues Concat(StringValues values1, StringValues values2)
+        {
+            return Equals(left, new StringValues(right));
+        }
+
+        public bool Equals(string other)
+        {
+            return Equals(this, new StringValues(other));
+        }
+
+        public static bool Equals(string[] left, StringValues right)
+        {
+            return Equals(new StringValues(left), right);
+        }
+
+        public static bool Equals(StringValues left, string[] right)
+        {
+            return Equals(left, new StringValues(right));
+        }
+
+        public bool Equals(string[] other)
+        {
+            return Equals(this, new StringValues(other));
+        }
+
+        public static bool operator ==(StringValues left, string right)
+        {
+            return Equals(left, new StringValues(right));
+        }
+
+        public static bool operator !=(StringValues left, string right)
+        {
+            return !Equals(left, new StringValues(right));
+        }
+
+        public static bool operator ==(string left, StringValues right)
+        {
+            return Equals(new StringValues(left), right);
+        }
+
+        public static bool operator !=(string left, StringValues right)
+        {
+            return !Equals(new StringValues(left), right);
+        }
+
+        public static bool operator ==(StringValues left, string[] right)
+        {
+            return Equals(left, new StringValues(right));
+        }
+
+        public static bool operator !=(StringValues left, string[] right)
+        {
+            return !Equals(left, new StringValues(right));
+        }
+
+        public static bool operator ==(string[] left, StringValues right)
+        {
+            return Equals(new StringValues(left), right);
+        }
+
+        public static bool operator !=(string[] left, StringValues right)
+        {
+            return !Equals(new StringValues(left), right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is string)
+            {
+                return Equals(this, (string)obj);
+            }
+            
+            if (obj is string[])
+            {
+                return Equals(this, (string[])obj);
+            }
+
+            if (obj is StringValues)
+            {
+                return Equals(this, (StringValues)obj);
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            if (_values == null)
+            {
+                return _value == null ? 0 : _value.GetHashCode();
+            }
+
+            switch (_values.Length)
+            {
+                case 0:
+                    return 0;
+
+                case 1:
+                    return _values[0].GetHashCode();
+
+                default:
+                    var hcc = new HashCodeCombiner();
+                    for (var i = 0; i < _values.Length; i++)
+                    {
+                        hcc.Add(_values[i].GetHashCode());
```

Comment:
You don't need to call `GetHashCode` here, the combiner does that for you - this is going to hash the string and then hash the hash